clients/admin-ui/src/pages/dataset/index.tsx

```diff
@@ -27,12 +27,14 @@ const DataSets: NextPage = () => {
   const toast = useToast();
 
   const handleLoadDataset = () => {
-    if (activeDatasetFidesKey) {
-      router.push(`/dataset/${activeDatasetFidesKey}`);
-      toast(successToastParams("Successfully loaded dataset"));
-    }
+    router.push(`/dataset/${activeDatasetFidesKey}`);
+    toast(successToastParams("Successfully loaded dataset"));
   };
 
+  if (activeDatasetFidesKey) {
+    handleLoadDataset();
+  }
```

Comment:
So I think what's going on is that this line will be called multiple times because of the way React renders things. It pretty much rerenders this component in the DOM every time anything changes. Using things like `useEffect` or `useMemo` help with this case, so it's possible this could be changed to

```typescript
useEffect(() => {
  if (activeDatasetFidesKey) {
    router.push(...)
    toast(...)
  }
}, [activeDatasetFidesKey])
```

I haven't tested that though. The safer thing to do is to directly call `handleLoadDataset` [when the table row is clicked](https://github.com/ethyca/fides/blob/main/clients/admin-ui/src/features/dataset/DatasetTable.tsx/#L28-L35) in the table component. I think that'd make more sense/is less risky (it's always kind of hard to track down React renders and why things are rerendering. much easier to track when something is the result of a button click!). So in addition to dispatching the `activeDatasetFidesKey` in `handleRowClick`, also do the `router.push` and `toast` there.

Let me know if that makes sense or if it'd be helpful to pair!

Reply:
Thank you for the explanation!! I think that makes a lot of sense and should give me enough to work back through :) will take you up on the pairing offer if I get stuck too much longer tho, thanks!!

Reply:
Thank you again @allisonking !! Think I have this going thanks to your help. I'd tried moving `handleLoadDataset` itself into `handleRowClick` earlier today and gotten an invalid hook call error - doing the `router.push` and `toast` explicitly in there instead solved it all 🙇🏽 thank you again for the quick and clear help with this!!